interview-questions.md

```diff
@@ -8,50 +8,73 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is the difference between a parameter and an argument?
 
-Your answer:
+Your answer: A parameter is a inputed value, index, or number in a function
+such as (array[bean, fruit, meat], /string"Hello How are you", index[2], 25) 
+where a argument an argument is if a value is given under a condition and then 
+it is logged in result as a return
+
+Researched answer: Parameters are the names listed in the functions Definition
+                    Example : function example(paramenter) {
+                        console.log(parameter)
+                    }
+                    An Argument are real values passed to the function
+                    const argument = "LEARN"
 
-Researched answer:
 
 2. The JavaScript built in method .map() takes predefined parameters. What are they? Which are required and which are optional?
 
-Your answer:
+Your answer: .map goes through each value in a array from a function and does not change the original array key and values are the predefined parameters
 
-Researched answer:
+Researched answer: .map is a collections of key value pairs. It does not contain any keys by default. it only contains what is explicitly put into it. It can be any value like functions objecs or any primitive. It iterates in the order of insertion.
 
 3. What is the difference between map and filter?
 
-Your answer:
+Your answer:map returns the information or log from the original array , where as filter returns a specific information based on what is being requested
 
-Researched answer:
+Researched answer:Filter creates an array filled with all elements pass a test implemented by the function and search for specifics through a function and logs those specifics back . 
+Such as finding fruits in a array of vegetables and fruits. 
+Map creates a new array going through EACH element in a function like dividing all numbers by 2 and logging those answers in its return.
 
 1. What is iteration?
 
-Your answer:
+Your answer:Iteration is process in a function where information or data is repeated and is returned what is requested in that function such as for loops.
 
-Researched answer:
+Researched answer:Iterative refers to a sequence of instructions or code being repeated until a specific end result is achieved. Iterative development is sometimes called circular or evolutionary development. const variable = (function) => {
+                                            for (string) =>{
```

Comment:
Researched answers need to be in your own words! We want to know what you understand and not what we can find on the internet and copy/paste.